src/main/java/nextstep/courses/service/RegistrationService.java

```diff
@@ -0,0 +1,29 @@
+package nextstep.courses.service;
+
+import nextstep.courses.domain.registration.Registration;
+import nextstep.courses.domain.registration.RegistrationRepository;
+import nextstep.courses.domain.session.Session;
+import nextstep.courses.domain.session.SessionRepository;
+import nextstep.payments.domain.Payment;
+import org.springframework.stereotype.Service;
+
+@Service
+public class RegistrationService {
+
+    private final SessionRepository sessionRepository;
+    private final RegistrationRepository registrationRepository;
+
+    public RegistrationService(SessionRepository sessionRepository,
+                               RegistrationRepository registrationRepository) {
+        this.sessionRepository = sessionRepository;
+        this.registrationRepository = registrationRepository;
+    }
+
+    public void register(Payment payment) {
+        Session session = sessionRepository.findById(payment.getSessionId());
+        session.validateEnroll(payment.getAmount());
+
+        Registration registration = new Registration(payment.getSessionId(), payment.getNsUserId());
+        registrationRepository.save(registration);
```

Comment:
Entity 객체를 생성하고 있는데 Repository에서 사용하는 이유는?

```suggestion
        SessionEntity session = sessionRepository.findById(payment.getSessionId());
        List<Registration> registrations = registrationRepository.findBySessionId(payment.getSessionId());
        Enrollment enrollment = session.enrollment(registrations);
        enrollment.enroll(payment);
```

위와 같이 Repository에 값을 전달하고 조회할 때 Entity를 사용하는 것은 어떻게 생각하나?

Reply:
(바꿔야지 하고 잊어버렸습니다) 
SessionRepository는 도메인 레이어에 있으므로 Session으로 받는게 맞는 거 같고
구현체는 인프라 레이어에 있으므로 SessionEntity를 사용하게 하려고 합니다